core/src/main/java/org/apache/spark/shuffle/api/metadata/ShuffleOutputTracker.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.shuffle.api.metadata;
+
+/**
+ * :: Private ::
+ *
+ * A plugin that can monitor the storage of shuffle data from map tasks, and can provide
+ * metadata to shuffle readers to aid their reading of shuffle blocks in reduce tasks.
+ * <p>
+ * {@link MapOutputMetadata} instances provided from the plugin tree's implementation of
+ * {@link org.apache.spark.shuffle.api.ShuffleMapOutputWriter} are sent to
+ * <p>
+ * Implementations MUST be thread-safe. Spark will invoke methods in this module in parallel.
+ */
+public interface ShuffleOutputTracker {
+
+  /**
+   * Called when a new shuffle stage is going to be run.
+   */
+  void registerShuffle(int shuffleId);
+
```

Comment:
A callback to signal completion (or failure) of shuffle for a `shuffleId` will help `ShuffleDriverComponents` coordinate with external services to commit (or cleanup) shuffle metadata/files.

Reply:
I'm hesitant to make significant API changes given how late this is in the review stage - let's add it as a follow-up if there are concrete use cases that require it down the road.

Reply:
Looking at the currently designed api, this is a missing gap.

Reply:
Again - I don't want to add it here, since it would require further integration of such a new API call in the rest of the Spark codebase, which I am holding as out of scope for this patch.

Reply:
This is as a gap in the proposed api - which limits the effectiveness to leverage it in for other shuffle usecases.
+CC @otterc PTAL if without this, shuffle changes can be made for your patches.

Reply:
This API isn't complete as of this patch anyways, so I'd prefer that functionality to be deferred since the complexity being added here is already pretty significant. Can we file a follow up JIRA for it and go from there?